### ISSUE TITLE
Make jenv for Windows have same commandline options and logic as https://www.jenv.be/

### ISSUE THREAD

**Issue by tech-consortium** [FelixSelter/JEnv-for-Windows#45]
There is already a very popular Jenv for linux and Windows.   If you want to get good adoption of this project I'd make the logic match https://www.jenv.be/ as much as possible.

**Comment by tech-consortium** [comment]
I'd also publish it to Chocolate package manager and other once you have the equivalent functionality as https://www.jenv.be/

**Comment by kid1412621** [comment]
I agree, at least use `jenv global xxx` instead of `jenv change xxx`, and `jenv shell xxx` instead of `jenv use xxx`.

**Comment by row49382** [comment]
Is there anyway to add the behavior of 'enable-plugin maven/gradle' that jenv has? 

I run multiple maven projects all in different java versions at work. It would be great to be able to run the mvn commands at different jdk levels that were set by jenv local from cmd line without needing to reset the JAVA_HOME value.

**Comment by FelixSelter** [comment]
If I understand you correctly we can easily create this. Currently jenv removes any java from your path and creates an own executable(batch file) named java or javac etc that ensures the correct version is used. We could create an mvn.bat file that calls maven and passes the correct version as an argument.

Alternatively you can just run jenv once after opening the terminal in that directiory. This should update your java home automatically.

Is this what you are looking for? I might need to look at the code if the second version is still working.

**Comment by row49382** [comment]
I think you understand it correctly.

Say I want java 17 globally and 8 locally. This is what happens currently:

* install java 8,17
* jenv add jdk8, jdk17
* jenv change jdk17
* cd "java_8_project_with_maven" 
* jenv local jdk8
* mvn clean install -> fails because requires target of java 8

> Alternatively you can just run jenv once after opening the terminal in that directiory. This should update your java home automatically.

Ah ok, so after swapping to "java_8_project_with_maven" and running jenv, then the version is registered correctly with maven and I am able to run install successfully. 

It's probably not worth the work to save the extra step, (though it would help usability by not needing to remember to call jenv each time I switch projects. 

Perhaps though, it is worth a callout in the wiki? I'll let you decide :) Thank you for the quick response!

**Comment by FelixSelter** [comment]
I will implement this in the full rewrite. at least once I have time. I am currently quite busy :(

**Comment by rcsilva83** [comment]
@FelixSelter, do you think jEnv and jEnv projects could merge to make one multiplatform jEnv? I've been thinking about the duplicate effort and, now that I know you're thinking about a full rewrite, I wonder if you and the jEnv team would be interested in this... Of course, this new jEnv shouldn't be written in PowerShell nor Shell Script...

**Comment by rcsilva83** [comment]
I'm not familiar with Python Virtual Environments but, if I understood right, in your solution the user would need to execute some kind of `setup` command once jEnv is installed, is it? I'd be something like [`volta setup`](https://docs.volta.sh/reference/setup) from Volta? I like this approach very much as it would make jEnv's faster and more predictable.

Nevertheless, my main point was the "multiplatform" aproach that you are into too, so I think we're on the same page :)

About uniting the efforts with the jenv team, let me ask them if they're interested. If not, I think it would be better a new name for the software, right? "jEnv for Windows" wouldn't work anymore.

**Comment by rcsilva83** [comment]
@FelixSelter , the jenv team [answered](https://github.com/jenv/jenv/issues/398#issuecomment-1579107104) they are open for a full rewrite to make a multiplatform tool. They asked which language do we plan to do it. What do you have in mind? As jEnv is a Java-related tool, I think a Java-related language (Java, Kotlin etc) would be appropriate, what do you think?

**Comment by FelixSelter** [comment]
The problem with a java related language is that it is compiled for a specific jvm version. Our tool changes the jvm version thus could break the executable.

 Personally I would use rust for cross platform support, security and speed. But writing a full scale application takes more time than with other languages. Maybe another language should be used then. I would like to avoid that users need to install any other runtime like python to run the code. I would prefer a statically typed language. Do you know any other language thats secure, supports cross platform and does not depend on an additional runtime?
If we decide to use a jvm language and risk that the runtime is moved or deleted I would prefer scala.

I actually created a prototype in rust already. If you would give me a couple more days I could improve documentation and create a repo?

**Comment by rcsilva83** [comment]
> The problem with a java related language is that it is compiled for a specific jvm version.

We could create a [native image with GraalVM](https://www.graalvm.org/native-image/) or bundle a mini-JVM created with [jlink](https://dev.java/learn/jvm/tools/core/jlink/).

**Comment by rcsilva83** [comment]
I opened an issue with some points we've discussed here: https://github.com/FelixSelter/JVenv/issues/1
